### DISCUSSION TITLE
V4.1 or stick with V4-2022-12-20?

### DISCUSSION THREAD

**Discussion by crazynightgriffin** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#467] [Q&A]
I have just got my St Moritz (square pump) working with V4 (i think dated 2022-12-20) of the firmware using a wemos D1 mini to discover v4.10 is available which seems a big enough jump to go for it while the pump is in pieces.

Has anyone got 4.10 working on a Wemos D1?

Cheers

PS Love the project and will be buying the dev a coffee.

**Comment by visualapproach** [comment]
Hey @crazynightgriffin looking forward to it 👍 thx

It should work with the D1 also. I just checked with google that the pinout to GPIO mapping is the same. So I'd go for it. After you install 4.1.0 you should be able to update to future firmware with the web ui. I've tested it myself, but since it's new I haven't heard if it works fine for others.

**Reply by crazynightgriffin** [reply]
I am going for it found a second wemos which has taken the project so just need to solder some headers on it and switch it out, means if it does not work can switch back to the original although now I know the screw would have been accessible without removing from the spa am a little less stressed.

Will keep you posted.

**Reply by visualapproach** [reply]
Yes, remember to go into hardware config page and set your model from the UI. Best to do before connecting to pump, but it should work either way.

**Reply by crazynightgriffin** [reply]
> Yes, remember to go into hardware config page and set your model from the UI. Best to do before connecting to pump, but it should work either way.

Its sorta working but cant matchup the instructions I followed with the pinouts on the hardware config page, keep getting strange results.  does the order in the old config match that of the web gui?
<img width="374" alt="image" src="https://user-images.githubusercontent.com/66025008/225103521-b886f0ee-ea08-49cb-844d-30f4e0a62902.png">

**Reply by visualapproach** [reply]
As I recall it is the same order

**Reply by visualapproach** [reply]
And you have to restart after setting the pins.

**Reply by crazynightgriffin** [reply]
Thanks for the help seems to be working with the following settings on our St Moritz
<img width="438" alt="image" src="https://user-images.githubusercontent.com/66025008/225107945-906ed2b2-72b7-419e-a7f8-15c22e27b602.png">

Not pretty but working.
<img width="387" alt="image" src="https://user-images.githubusercontent.com/66025008/225108580-19f56b92-801e-4d42-97f7-d3f3601c2194.png">

<img width="358" alt="image" src="https://user-images.githubusercontent.com/66025008/225108066-3d688592-7cd8-4515-8b5d-80c52500bfaa.png">
<img width="137" alt="image" src="https://user-images.githubusercontent.com/66025008/225108137-d7722e2e-a519-4d6a-b11f-e4b8b9752a83.png">

**Reply by visualapproach** [reply]
Sweet!
